src/components/track-changes/TrackChangesPanel.tsx

```diff
@@ -10,179 +10,112 @@
  * All portions of the code written by Atypon Systems LLC are Copyright (c) 2019 Atypon Systems LLC. All Rights Reserved.
  */
 
-import { SET_SUGGESTION_ID } from '@manuscripts/body-editor'
 import {
   CHANGE_STATUS,
   ChangeSet,
   trackCommands,
   TrackedChange,
 } from '@manuscripts/track-changes-plugin'
-import { NodeSelection, Selection, TextSelection } from 'prosemirror-state'
-import React, { useEffect, useState } from 'react'
+import { NodeSelection, TextSelection } from 'prosemirror-state'
+import React, { useState } from 'react'
 
 import useExecCmd from '../../hooks/use-exec-cmd'
 import { useStore } from '../../store'
 import { SnapshotsDropdown } from '../inspector/SnapshotsDropdown'
 import { SortByDropdown } from './SortByDropdown'
 import { SuggestionList } from './suggestion-list/SuggestionList'
 
-export function TrackChangesPanel() {
+export const TrackChangesPanel: React.FC = () => {
   const [sortBy, setSortBy] = useState('in Context')
 
-  const [{ editorSelectedSuggestion, editor, trackState }, dispatch] = useStore(
-    (store) => ({
-      editorSelectedSuggestion: store.editorSelectedSuggestion,
-      editor: store.editor,
-      trackState: store.trackState,
-    })
-  )
-
-  const execCmd = useExecCmd()
+  const [{ editor, trackState, selectedSuggestionID }] = useStore((store) => ({
+    editor: store.editor,
+    trackState: store.trackState,
+    selectedSuggestionID: store.selectedSuggestionID,
+  }))
 
-  const { changeSet } = trackState || {}
-
-  const cleanTextSelection = () => {
-    const { view, dispatch } = editor
-    if (view && view.state.selection instanceof TextSelection) {
-      view.focus()
-      const tr = view.state.tr.setSelection(
-        Selection.near(view.state.doc.resolve(view.state.selection.anchor))
-      )
-      tr.setMeta('CLEAR_SUGGESTION_ID', true)
-      dispatch(tr)
+  const setSelectedSuggestion = (suggestion: TrackedChange) => {
+    const state = editor.state
+    const view = editor.view
+    const tr = state.tr
+    if (suggestion.type === 'text-change') {
+      const pos = suggestion.to - 1
+      tr.setSelection(TextSelection.create(state.doc, pos, pos))
+    } else {
+      tr.setSelection(NodeSelection.create(state.doc, suggestion.from))
     }
+    view?.dispatch(tr.scrollIntoView())
```

Comment:
will need to also add `view.focus()` for that to work when the selection is out of editor

Reply:
Added.